### ISSUE TITLE
`$ fontknife convert tom-thumb.{bdf,png}`: ValueError: x1 must be greater than or equal to x0

### ISSUE THREAD

**Issue by Des-Nerger** [pushfoo/fontknife#4]
```
$ fontknife convert tom-thumb.{bdf,png}
Traceback (most recent call last):
  File "/home/username/.local/bin/fontknife", line 8, in <module>
    sys.exit(main())
             ^^^^^^
  File "/home/username/.local/lib/python3.12/site-packages/fontknife/commands/__main__.py", line 361, in main
    font = load_font(source_path, **source_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/username/.local/lib/python3.12/site-packages/fontknife/formats/__init__.py", line 84, in load_font
    font = reader.load_source(source, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/username/.local/lib/python3.12/site-packages/fontknife/formats/common/__init__.py", line 243, in load_source
    result = RasterFont(**raw_glyph_data, size_points=font_size, path=actual_path)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/username/.local/lib/python3.12/site-packages/fontknife/formats/common/raster_font.py", line 195, in __init__
    self._update_max_tile_size_and_tofu()
  File "/home/username/.local/lib/python3.12/site-packages/fontknife/formats/common/raster_font.py", line 231, in _update_max_tile_size_and_tofu
    self._notdef_glyph = generate_missing_character_core(self._max_tile_bbox[:2])
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/username/.local/lib/python3.12/site-packages/fontknife/formats/common/raster_font.py", line 149, in generate_missing_character_core
    draw.rectangle(rectangle_bbox, outline=color)
  File "/home/username/.local/lib/python3.12/site-packages/PIL/ImageDraw.py", line 296, in rectangle
    self.draw.draw_rectangle(xy, ink, 0, width)
ValueError: x1 must be greater than or equal to x0
```

**Comment by pushfoo** [comment]
Ty for the report! I haven't been able to replicate this yet. Can you please provide the following?

1. How you installed Fontknife?
2. Your OS info
3. `python --version` output
4. Any other pip / venv info

### What I've tried
I tried the following approaches on a fresh 3.12 venv under Linux with Python built from source.

#### From `pip`
`pip install fontknife` breaks on 3.12 due to Pillow wheel issues. Thank you for bringing this to my attention.

#### From source
I ran the command you provided as follows with no errors:
1. `cd` into repo root
2. Create a fresh 3.12 venv with `python -m venv .venv3.12`
3. `source .venv3.12/bin/activate`
4. `pip install -e .`
5. `fontknife convert tom-thumb.{bdf,png}`

**Comment by Des-Nerger** [comment]
> provide the following
1. `$ pip install --user --break-system-packages fontknife`
Quite a risky approach, yes. But it never failed me, so I kept using it.

2.
```sh
$ head -n1 /etc/os-release; uname -srvmo
PRETTY_NAME="Debian GNU/Linux trixie/sid"
Linux 6.11.10-amd64 #1 SMP PREEMPT_DYNAMIC Debian 6.11.10-1 (2024-11-23) x86_64 GNU/Linux
```

3.
```sh
$ python3 --version
Python 3.12.7
```

4.
```sh
$ pip --version
pip 24.3.1 from /usr/lib/python3/dist-packages/pip (python 3.12)
```
It was installed from the package `python3-pip` in the official Debian repos.

**Comment by pushfoo** [comment]
What Pillow version are you running this against? My guess is it's a different one than it expects.

> Quite a risky approach, yes. But it never failed me, so I kept using it.

[`pipx`](https://github.com/pypa/pipx) is worth considering. It's meant specifically for use cases like this one.

**Comment by Des-Nerger** [comment]
```bash
$ grep -Ei '^(__)?version[_:]' ~/.local/lib/python3.12/site-packages/{PIL/_version.py,Pillow-9.5.0.dist-info/METADATA}
/home/username/.local/lib/python3.12/site-packages/PIL/_version.py:__version__ = "9.5.0"
/home/username/.local/lib/python3.12/site-packages/Pillow-9.5.0.dist-info/METADATA:Version: 9.5.0
```

**Comment by pushfoo** [comment]
**TL;DR:** Confirmed with an easy temp fix till I get a release out.

Ty for the report! The temp workaround's easy as long as you include the quotes:
```
pip install "fontknife @ https://github.com/pushfoo/Fontknife/zipball/4979a00576c890d70763de055f8bf0b90a57c239"
```
There's a better, shorter syntax but:
1. I'm on mobile at the moment
2. I had this on hand autogenerated in the doc

**Comment by Des-Nerger** [comment]
Thanks, it works!
I was able to easily extract Terminus glyphs into a nice .png, as I originally wanted when installed Fontknife.
![ter-u24n](https://github.com/user-attachments/assets/b49e1dfa-a005-4bb1-9b34-5ac2495d597e)
